cpp/src/cider/common/base/StringRef.h

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright(c) 2022-2023 Intel Corporation.
+ * Copyright (c) 2016-2022 ClickHouse, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <cassert>
+#include <functional>
+#include <iosfwd>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <common/base/unaligned.h>
+#include <common/contrib/cityhash102/include/city.h>
+
+#include <emmintrin.h>
+#include <nmmintrin.h>
+#include <smmintrin.h>
+
+namespace cider::hashtable {
+
+/**
+ * The std::string_view-like container to avoid creating strings to find substrings in the
+ * hash table.
+ */
+struct StringRef {

```

Comment:
I think ``` class StringRef``` will be better

Reply:
no difference with std::string_view?

Reply:
It differs when the size of string changes. In substring cases, what we need is to change the size and keep the data pointer instead of creating a new string.